vault/token_store_test.go

```diff
@@ -832,6 +832,35 @@ func TestTokenStore_HandleRequest_CreateToken_NonRoot_InvalidSubset(t *testing.T
 	}
 }
 
+func TestTokenStore_HandleRequest_CreateToken_NonRoot_RootChild(t *testing.T) {
+	_, ts, _, root := TestCoreWithTokenStore(t)
+	testMakeToken(t, ts, root, "sudoClient", "", []string{"create", "update", "sudo"})
+
+	req := logical.TestRequest(t, logical.UpdateOperation, "create")
+	req.ClientToken = "sudoClient"
+	req.Data["policies"] = []string{"root", "create"}
+
+	resp, err := ts.HandleRequest(req)
+	if err != logical.ErrInvalidRequest {
+		t.Fatalf("err: %v %v", err, resp)
+	}
```

Comment:
You should ensure that `resp` is not nil, and then that `resp.Data` is not nil, before proceeding, as this will lead to a panic otherwise.